### PR TITLE
YCSB: allow arbitrary initial table size; use the same set of keys fo…

### PR DESCRIPTION
**Pull request by wangtzh**
…r retries.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/hkimura/foedus_code/pull/113%23discussion_r45427928%22%2C%20%22https%3A//github.com/hkimura/foedus_code/pull/113%23issuecomment-158255874%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/hkimura/foedus_code/pull/113%23issuecomment-158255874%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Found%20no%20blocking%20thing.%20Merging.%22%2C%20%22created_at%22%3A%20%222015-11-20T01%3A51%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/702382%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/hkimura%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2037a548c3db78a1ba461fb4dffee8d74c72469b93%20experiments-core/src/foedus/ycsb/ycsb_client.cpp%20123%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/hkimura/foedus_code/pull/113%23discussion_r45427928%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20forgot%2C%20but%20is%20this%20assertion%20always%20correct%3F%22%2C%20%22created_at%22%3A%20%222015-11-20T01%3A50%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/702382%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/hkimura%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20experiments-core/src/foedus/ycsb/ycsb_client.cpp%3AL170-211%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 37a548c3db78a1ba461fb4dffee8d74c72469b93 experiments-core/src/foedus/ycsb/ycsb_client.cpp 123'></a> &nbsp;<b><a href='https://github.com/hkimura/foedus_code/pull/113#discussion_r45427928'>File: experiments-core/src/foedus/ycsb/ycsb_client.cpp:L170-211</a></b>
- <a href='https://github.com/hkimura'></a> I forgot, but is this assertion always correct?
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/hkimura/foedus_code/pull/113#issuecomment-158255874'>General Comment</a></b>
- <a href='https://github.com/hkimura'></a> Found no blocking thing. Merging.

<a href='https://www.codereviewhub.com/hkimura/foedus_code/pull/113?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/hkimura/foedus_code/pull/113?approve=1'></a>&nbsp;<a href='https://github.com/hkimura/foedus_code/pull/113'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by hkimura** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/hkimura/foedus_code/pull/113'>Refresh</a>.<a href='#crh-update-none'></a>

File: experiments-core/src/foedus/ycsb/ycsb_client.cpp [hunk 1]
```diff
@@ -152,33 +170,41 @@ ErrorStack YcsbClientTask::run(thread::Thread* context) {
...
-
-      if (context->is_running_xct()) {
+        ret = xct_manager_->precommit_xct(context_, &commit_epoch);
+        if (ret == kErrorCodeOk) {
+          ASSERT_ND(!context->is_running_xct());
+          access_keys.clear();
+          break;
+        }
+      } else {
+        ASSERT_ND(context->is_running_xct());
```
Comment (hkimura) [inline experiments-core/src/foedus/ycsb/ycsb_client.cpp]: I forgot, but is this assertion always correct?

**Comment by hkimura** [comment]
Found no blocking thing. Merging.

File: experiments-core/src/foedus/ycsb/ycsb_client.cpp (hunk 1 above)
Comment (wangtzh) [inline experiments-core/src/foedus/ycsb/ycsb_client.cpp]: I think so, this should be the only place we do abort_xct and precommit_xct?

**Pull request merged**